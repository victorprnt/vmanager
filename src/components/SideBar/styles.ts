import styled from 'styled-components'

export const Wrapper = styled.nav`
  display: flex;
  flex-direction: column;
  align-items: center;

  position: absolute;
  top: 0;
  left: 0;

  height: 100vh;
  max-width: 4vw;
  background-color: #202020;

  transition: all 0.3s ease-in-out;

  &:hover {
    max-width: 10vw;

    .page-link-container {
      justify-content: flex-start;

      .page-link {
        .page-link-text {
          display: flex;
        }
      }
    }
  }

  .logo-container {
    max-width: 60px;
    width: 100%;
    margin-bottom: 1.5rem;

    img {
      width: 100%;
      padding: 1rem;
      color: #fff;
    }
  }

  .page-link-container {
    display: flex;
    justify-content: center;
    width: 100%;
    margin-bottom: 3rem;

    /* padding: 1rem; */

    transition: all 0.3s ease-in-out;

    &:hover {
      background-color: #303030;
    }

    .page-link {
      display: flex;
      /* flex-direction: column; */
      font-size: 1.5rem;

      img {
        width: 28px;
      }

      .page-link-text {
        display: none;
        margin-left: 0.75rem;

        transition: all 0.3s ease-in-out;
      }
    }
  }
`
