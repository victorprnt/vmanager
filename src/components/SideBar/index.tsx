import * as S from './styles'

const SideBar = () => (
  <S.Wrapper>
    <h1>SideBar</h1>
  </S.Wrapper>
)

export default SideBar
