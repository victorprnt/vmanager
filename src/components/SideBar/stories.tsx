// import { Story, Meta } from '@storybook/react'
import SideBar from '.'

// export default {
//   title: 'SideBar',
//   component: SideBar
// } as Meta

// export const Default: Story = () => <SideBar />
