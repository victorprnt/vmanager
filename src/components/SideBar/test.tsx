import { render, screen } from '@testing-library/react'

import SideBar from '.'

describe('<SideBar />', () => {
  it('should render the heading', () => {
    const { container } = render(<SideBar />)

    expect(screen.getByRole('heading', { name: /SideBar/i })).toBeInTheDocument()

    expect(container.firstChild).toMatchSnapshot()
  })
})
