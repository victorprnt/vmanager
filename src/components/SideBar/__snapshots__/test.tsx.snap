// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<SideBar /> should render the left navbar 1`] = `
<nav
  class="sc-bczRLJ eqBENw"
>
  <div
    class="logo-container"
  >
    <img
      alt="mph logo"
      class="logo"
      src="mph-logo.png"
    />
  </div>
  <div
    class="page-link-container page-dashboard"
  >
    <a
      class="page-link"
      href="/dashboard"
    >
      <img
        alt="Dashboard image to lead to dashboard page"
        src="dashboard-line.svg"
        title="Dashboard link"
      />
      <p
        class="page-link-text"
      >
        Dashboard
      </p>
    </a>
  </div>
  <div
    class="page-link-container page-hosts"
  >
    <a
      class="page-link"
      href="/hosts"
    >
      <img
        alt="Table image to lead to the hosts page"
        src="computer-line.svg"
        title="Hosts link"
      />
      <p
        class="page-link-text"
      >
        Hosts
      </p>
    </a>
  </div>
  <div
    class="page-link-container page-vulnerabilities"
  >
    <a
      class="page-link"
      href="/vulnerabilities"
    >
      <img
        alt="Table image to lead to the vulnerability page"
        src="table-line.svg"
        title="Vulnerabilities link"
      />
      <p
        class="page-link-text"
      >
        Vulnerabilites
      </p>
    </a>
  </div>
</nav>
`;
